### PR TITLE
Prevent empty work items due to failed attachment import.

### PR DESCRIPTION
**Pull request by bameije**
Added try-catch around each attachment import. When the import fails, the containing work item will still be imported correctly. Example reason of failure: DevOps server can limit the maximum attachment file upload size.

Without this change, all remaining code in Agent.ImportRevision() would be skipped. Because the attachments are in the first revision of the work item, the created work item is then left empty. Importing the following revisions would then trigger another work item to be created, which would not contain title, description and all other fields that were defined only in the first revision.

See Issue #638

### CODE REVIEW

File: src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs [hunk 1]
```diff
@@ -467,13 +467,27 @@ public void SaveWorkItemAttachments(WiRevision rev, WorkItem wi)
...
                 {
-                    RemoveSingleAttachmentFromWorkItemAndSave(attachment, wi);
+                    try
+                    {
+                        Logger.Log(ex, $"Failed to import attachment '{attachment.ToString()}'.");
+                    }
+                    catch (Exception)
+                    {
+                        throw ex; // yes, the ex from the nesting catch block
+                    }
```
Comment (Alexander-Hjelm) [inline src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs:490]: I'm not a huge fan of nested try-catch block. Why is this needed? What exception can be thrown from this line? `Logger.Log(ex, $"Failed to import attachment '{attachment.ToString()}'.")`
@bameije

**Review by Alexander-Hjelm — CHANGES_REQUESTED** [review]

**Comment by bameije** [comment]
It's my way of safely adding stuff to code that I don't know: if I used the
Logger.Log() in a wrong way (e.g. causing a null-ref exception or
whatever), I want the program to behave exactly as before. Thus, not
throwing the exception that I caused, but the original exception.

The code has proven to work fine for me, so indeed, if you're confident as
well that Logger.Log() won't throw then you can remove the inner try-catch.

Regards,
Brecht

File: src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs (hunk 1 above)
Comment (bameije) [reply src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs:490]: It's my way of safely adding stuff to code that I don't know: if I used the Logger.Log() in a wrong way (e.g. causing a null-ref exception or whatever), I want the program to behave exactly as before. Thus, not throwing the exception that I caused, but the original exception.

As you have requested this change in the main branch, we don't need this safety check anymore. I'll remove it.
```suggestion
                    Logger.Log(ex, $"Failed to import attachment '{attachment.ToString()}'.");
```
